Index: JQuery.hx
===================================================================
--- JQuery.hx	(revision 54)
+++ JQuery.hx	(working copy)
@@ -24,9 +24,6 @@
 	
 	public function live(type:String, fn:Event -> Void):Void;
 	
-	public function uploadify(settings:Dynamic):Void;
-	public function uploadifyClearQueue():Void;
-	
     /**
         Core
     **/
Index: Main.hx
===================================================================
--- Main.hx	(revision 54)
+++ Main.hx	(working copy)
@@ -1,33 +1,7 @@
-/*
- * Copyright (c) 2008, TouchMyPixel & contributors
- * Original author : Tony Polinelli
- * Contributors:
- * All rights reserved.
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *
- *   - Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   - Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *
- * THIS SOFTWARE IS PROVIDED BY THE TOUCH MY PIXEL & CONTRIBUTERS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE TOUCH MY PIXEL & CONTRIBUTORS
- * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
- * THE POSSIBILITY OF SUCH DAMAGE.
- */
-
 package;
 
-import php.db.Mysql;
+import php.Session;
+import php.Sys;
 import poko.Application;
 import poko.utils.PhpTools;
 import haxe.Timer;
@@ -35,66 +9,26 @@
 import php.Web;
 
 import site.cms.ImportAll;
-import site.examples.ImportAll;
 
 
-import site.services.Rss;
-import site.services.XmlData;
+import site.wholesale.Home;
+import site.wholesale.Display;
+import site.Schools;
+
 import site.services.Image;
+import site.services.Download;
 
-import site.Index;
-
-/*
-class Test {
+class Test implements Dynamic
+{
 	
-	public function new() 
+	public function new()
 	{
-		//Hash
-		var hash = new Hash();
-		hash.set("a", 1);
-		hash.set("b", 2);
-		for (item in hash.keys())
-			trace(item);
-		
-		for (item in hash)
-			trace(item);
-			
-		//HList
-		//_hx_list_iterator
-		var list = new List();
-		list.add("a");
-		list.add("b");
-		
-		for (item in list)
-			trace(item);
-		
-		
-		//_hx_array
-		//_hx_array_iterator
-		var array = ["a", "b"];
-		for (item in array)
-			trace(item);
-			
-		
-		//IntHash
-		var intHash = new IntHash();
-		intHash.set(1, "a");
-		intHash.set(2, "b");
-		for (item in intHash)
-			trace(item);
-
+		var str = "sdsds";		
+		trace(str.split(""));
 	}
-	
-	
-	
-	public function toString()
-	{
-		return "toString on Test1";
-	}
+
 }
-*/
 
-
 class Main 
 {
 	static private var app:Application;
@@ -109,7 +43,6 @@
 		PhpTools.setupTrace();
 		
 		
-	//	Type.createInstance(Type.resolveClass("Test"),[]);
 		
 		app = new Application();
 		
@@ -122,9 +55,10 @@
 		
 		app.skipAuthentication = false;		
 		app.debug = true;
-		app.sessionName = "pokocms";
+		app.sessionName = "dobsons_wholesale";
 			
-		
+		app.defaultRequest = "Home";
+				
 		if (app.debug) 
 		{
 			execute(phpMode);

 			
Index: MainJS.hx
===================================================================
--- MainJS.hx	(revision 54)
+++ MainJS.hx	(working copy)
@@ -1,38 +1,13 @@
-/*
- * Copyright (c) 2008, TouchMyPixel & contributors
- * Original author : Tony Polinelli
- * Contributors:
- * All rights reserved.
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *
- *   - Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   - Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *
- * THIS SOFTWARE IS PROVIDED BY THE TOUCH MY PIXEL & CONTRIBUTERS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE TOUCH MY PIXEL & CONTRIBUTORS
- * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
- * THE POSSIBILITY OF SUCH DAMAGE.
- */
 
 package;
-
 import js.Dom;
 import js.Lib;
 import poko.js.JsApplication;
 
 import site.cms.ImportAll;
 
+import site.js.JsSchools;
+
 class MainJS 
 {
 	static private var app:JsApplication;
@@ -40,7 +15,7 @@
 	public static function main() 
 	{
         app = new JsApplication();
-		app.serverUrl = "http://localhost/poko/";
+		app.serverUrl = "http://localhost/fwork/";
 		
 		js.Lib.window.onload = run;
 	}
Index: poko/Application.hx
===================================================================
--- poko/Application.hx	(revision 54)
+++ poko/Application.hx	(working copy)
@@ -73,8 +73,6 @@
 	
 	public var errorContactEmail:String;
 	
-	public var components:List<Component>;
-	
 	public function new() 
 	{
 		/**
@@ -107,8 +105,6 @@
 		
 		db = new Db();
 		isDbRequired = true;
-		
-		components = new List();
 	}
 	
 	public function setupRequest():Void
@@ -151,8 +147,7 @@
 	
 	private function setupSessionData():Void
 	{
-		Session.setName(sessionName);
-		
+		//Session.setName(sessionName);
 		messages = Session.get("messages") ? Session.get("messages") : new Messages();
 		user = Session.get("user") ? Session.get("user") : new User();
 		user.update();
Index: poko/Component.hx
===================================================================
--- poko/Component.hx	(revision 54)
+++ poko/Component.hx	(working copy)
@@ -39,7 +39,6 @@
 		template_file = StringTools.replace(Type.getClassName(Type.getClass(this)), ".", "/") + ".mtt";
 		
 		application = Application.instance;
-		application.components.add(this);
 	}
 	
 	public function setTemplate(file:String):Void
@@ -47,7 +46,6 @@
 		template_file = Application.instance.packageRoot + "/" + file;
 	}
 	
-	public function pre(){}
 	public function main(){}
 	
 	public function setOutput(value)
@@ -57,6 +55,8 @@
 	
 	override public function render():String
 	{
+		main();
+		
 		return output != null ? output : super.render();
 	}
 	
Index: poko/form/elements/CheckboxGroup.hx
===================================================================
--- poko/form/elements/CheckboxGroup.hx	(revision 54)
+++ poko/form/elements/CheckboxGroup.hx	(working copy)
@@ -30,6 +30,7 @@
 import poko.form.Form;
 import poko.form.FormElement;
 import php.Web;
+import poko.form.Formatter;
 
 class CheckboxGroup extends FormElement 
 {
@@ -39,6 +40,9 @@
 	public var verticle:Bool;
 	public var labelRight:Bool;
 	
+	public var formatter:Formatter;
+	public var columns:Int;
+	
 	public function new(name:String, label:String, data:List<Dynamic>, ?selected:Array<String>, ?verticle:Bool=true, ?labelRight:Bool=true) 
 	{
 		super();
@@ -48,6 +52,8 @@
 		this.value = selected != null ? selected : new Array();
 		this.verticle = verticle;
 		this.labelRight = labelRight;
+		
+		columns = 4;
 	}
 	
 	override public function populate()
@@ -77,17 +83,51 @@
 		}
 		
 		var c = 0;
-		if (data != null)
+		var array = Lambda.array(data);
+		if (array != null)
 		{
-			for (row in data)
+			var rowsPerColumn = Math.ceil(array.length / columns);
+			s = "<table><tr>";
+			for (i in 0...columns)
 			{
-				var checkbox = "<input type=\"checkbox\" name=\""+n+"[]\" id=\""+n+c+"\" value=\"" + row.key + "\" " + (value != null ? Lambda.has(value, row.key+"") ? "checked":"":"") +" ></input>\n";
-				var label = "<label for=\"" + n+c + "\" >" + row.value  +"</label>";
+				s += "<td valign=\"top\">\n";
+				s += "<table>\n";
 				
-				s += labelRight ? checkbox + " "+label+" ": label+" "+checkbox+" ";
-				if (verticle) s += "<br />";
-				c++;
-			}	
+				for (j in 0...rowsPerColumn)
+				{
+					s += "<tr>";
+					
+					var row:Dynamic = array[c];
+					
+					var checkbox = "<input type=\"checkbox\" name=\""+n+"[]\" id=\""+n+c+"\" value=\"" + row.key + "\" " + (value != null ? Lambda.has(value, row.key+"") ? "checked":"":"") +" ></input>\n";
+					var label;
+					
+					if (formatter != null)
+					{
+						label = "<label for=\"" + n + c + "\" >" + formatter.format(row.value)  +"</label>\n";
+					} else {
+						label = "<label for=\"" + n + c + "\" >" + row.value  +"</label>\n";
+					}
+					
+					if (labelRight)
+					{
+						s += "<td>" + checkbox + "</td>\n";
+						s += "<td>" + label + "</td>\n";
+					} else {
+						s += "<td>" + label + "</td>\n";
+						s += "<td>" + checkbox + "</td>\n";
+					}
+					s += "</tr>";
+					
+					c++;
+					
+					if (c >= array.length) break;
+				}
+				s += "</table>";
+				s += "</td>";
+			}
+			s += "</tr></table>\n";
+			
 		}
 		
 		return s;
Index: poko/form/elements/Input.hx
===================================================================
--- poko/form/elements/Input.hx	(revision 54)
+++ poko/form/elements/Input.hx	(working copy)
@@ -32,6 +32,7 @@
 import poko.form.FormElement;
 import poko.form.Validator;
 import poko.form.validators.BoolValidator;
+import poko.form.Formatter;
 
 
 class Input extends FormElement
@@ -42,6 +43,8 @@
 	public var useSizeValues:Bool;
 	public var printRequired:Bool;
 	
+	public var formatter:Formatter;
+	
 	public function new(name:String, label:String, ?value:String, ?required:Bool=false, ?validators:Array<Validator>, ?attibutes:String="") 
 	{
 		super();
Index: poko/form/elements/RadioGroup.hx
===================================================================
--- poko/form/elements/RadioGroup.hx	(revision 54)
+++ poko/form/elements/RadioGroup.hx	(working copy)
@@ -38,7 +38,7 @@
 	public var verticle:Bool;
 	public var labelRight:Bool;
 	
-	public function new(name:String, label:String, data:List<Dynamic>, ?selected:String, defaultValue:String, ?verticle:Bool=true, ?labelRight:Bool=true) 
+	public function new(name:String, label:String, data:List<Dynamic>, ?selected:String, ?defaultValue:String, ?verticle:Bool=true, ?labelRight:Bool=true) 
 	{
 		super();
 		this.name = name;
Index: poko/form/Form.hx
===================================================================
--- poko/form/Form.hx	(revision 54)
+++ poko/form/Form.hx	(working copy)
@@ -37,6 +37,7 @@
 
 class Form
 {
+	public var id:String;
 	public var name:String;
 	public var action:String;
 	public var method:FormMethod;
@@ -50,7 +51,7 @@
 	public function new(name:String, ?action:String, ?method:FormMethod) 
 	{
 		this.forcePopulate = false;
-		this.name = name;
+		this.id = this.name = name;
 		this.action = action;
 		this.method = (method == null) ? FormMethod.POST : method;
 		
@@ -136,7 +137,7 @@
 	
 	public function getOpenTag():String
 	{
-		return "<form name=\"" + name + "\" method=\""+ method +"\" action=\""+ action +"\" enctype=\"multipart/form-data\" >";
+		return "<form id=\""+id+"\" name=\"" + name + "\" method=\""+ method +"\" action=\""+ action +"\" enctype=\"multipart/form-data\" >";
 	}
 	
 	public function getCloseTag():String
Index: poko/Request.hx
===================================================================
--- poko/Request.hx	(revision 54)
+++ poko/Request.hx	(working copy)
@@ -164,8 +164,6 @@
 		
 		pre();
 		
-		for (component in application.components) component.pre();
-		
 		if (!application.skipAuthentication && (authenticate && !auth()))
 		{
 			application.redirect("?request=cms.Index");
@@ -177,8 +175,6 @@
 			{
 				main();
 				
-				for (component in application.components) component.main();
-				
 				if (request_output == null) 
 				{
 					var tpl = "./tpl/" + StringTools.replace(template_file, "/", "__") + ".php";
@@ -228,6 +224,7 @@
 	public var publisher:String;
 	public var date:String;
 	public var favicon:String;
+	public var author:String;
 	
 	public var js:List<String>;
 	public var css:List<String>;
Index: poko/TemploContext.hx
===================================================================
--- poko/TemploContext.hx	(revision 54)
+++ poko/TemploContext.hx	(working copy)
@@ -78,7 +78,7 @@
 		// prepare dynamic properties (send in from constructor)
 		var k = "";
 		var v:Dynamic = null;
-		untyped __php__("foreach($this->»dynamics as $k=> $v){ ");
+		untyped __php__("foreach($this->»dynamics as $k=> $v){ ");		
 				
 			if (Std.is(v, TemploRenderable)) 
 			{
			
Index: site/cms/modules/base/Dataset.hx
===================================================================
--- site/cms/modules/base/Dataset.hx	(revision 54)
+++ site/cms/modules/base/Dataset.hx	(working copy)
@@ -30,6 +30,7 @@
 import haxe.Public;
 import php.Session;
 import poko.form.elements.Hidden;
+import poko.form.Formatter;
 import poko.js.JsBinding;
 import poko.form.elements.Input;
 import poko.form.elements.Button;
@@ -539,28 +540,33 @@
 	{
 		var data:Dynamic = Reflect.field(row, field);
 		var properties = definition.getElement(field).properties;
-		
-		return switch(properties.type)
+		if (properties.formatter != null && properties.formatter != "")
 		{
-			case "text": (data).substr(0,50) + (data.length > 50 ? "..." :  "");
-			case "richtext-tinymce": StringTools.htmlEscape(data.substr(0, 50)) + ((data.length > 50) ? "..." : "");
-			case "richtext-wym": StringTools.htmlEscape(data.substr(0, 50)) + ((data.length > 50) ? "..." : "");
-			case "image-file":
-				if (properties.isImage == "1"){
-					"<a target=\"_blank\" href=\"?request=cms.services.Image&src=" + data + "\"><img src=\"?request=cms.services.Image&preset=tiny&src=" + data + "\" /></a> <br/>";
-				}else {
-					if(data){
-						"<a target=\"_blank\" href=\"./res/uploads/" + data + "\" />file</a>";
+			var f:Formatter = Type.createInstance(Type.resolveClass(properties.formatter), []);
+			return f.format(data);
+		} else {
+			return switch(properties.type)
+			{
+				case "text": (data).substr(0,50) + (data.length > 50 ? "..." :  "");
+				case "richtext-tinymce": StringTools.htmlEscape(data.substr(0, 50)) + ((data.length > 50) ? "..." : "");
+				case "richtext-wym": StringTools.htmlEscape(data.substr(0, 50)) + ((data.length > 50) ? "..." : "");
+				case "image-file":
+					if (properties.isImage == "1"){
+						"<a target=\"_blank\" href=\"?request=cms.services.Image&src=" + data + "\"><img src=\"?request=cms.services.Image&preset=tiny&src=" + data + "\" /></a> <br/>";
 					}else {
-						"empty";
+						if(data){
+							"<a target=\"_blank\" href=\"./res/uploads/" + data + "\" />file</a>";
+						}else {
+							"empty";
+						}
 					}
-				}
-			case "bool": formatBool(cast data, properties);
-			case "date": formatDate(cast data);
-			case "keyvalue": "list of values";
-			case "association":
-				properties.showAsLabel == "1" ? associateExtras.get(field).get(cast data) : data;
-			default: data;
+				case "bool": formatBool(cast data, properties);
+				case "date": formatDate(cast data);
+				case "keyvalue": "list of values";
+				case "association":
+					properties.showAsLabel == "1" ? associateExtras.get(field).get(cast data) : data;
+				default: data;
+			}
 		}
 	}
 	public function formatBool(data:Bool, properties:Dynamic)
Index: site/cms/modules/base/DatasetItem.hx
===================================================================
--- site/cms/modules/base/DatasetItem.hx	(revision 54)
+++ site/cms/modules/base/DatasetItem.hx	(working copy)
@@ -35,7 +35,6 @@
 import site.cms.common.DefinitionElementMeta;
 import poko.form.elements.Button;
 import poko.form.elements.CheckboxGroup;
-import site.cms.components.PopupURL;
 import site.cms.modules.base.formElements.KeyValueInput;
 import poko.form.elements.DateSelector;
 import site.cms.modules.base.formElements.FileUpload;
@@ -79,8 +78,6 @@
 	
 	public var jsBind:JsBinding;
 	
-	public var dbselector:PopupURL;
-	
 	public function new() 
 	{
 		super();
@@ -116,6 +113,7 @@
 		dataset = Std.parseInt(application.params.get("dataset"));
 		isOrderingEnabled = false;
 		
+		
 		if (!pagesMode)
 		{
 			// Dataset Mode
@@ -447,7 +445,8 @@
 					}
 					
 					// print out element.properties.regexDescription so we know what we need to input (if it exists)
-					
+					if (element.properties.formatter != null && element.properties.formatter != "") 
+						el.formatter = Type.createInstance(Type.resolveClass(element.properties.formatter), []);
 					el.description = element.properties.description;
 					form.addElement(el);
 					
@@ -501,7 +500,7 @@
 					if (element.properties.mode) el.mode = Type.createEnum(Type.resolveEnum("poko.form.elements.RichtextMode"), element.properties.mode);
 					if (element.properties.width != "") el.width = Std.parseInt(element.properties.width);
 					if (element.properties.height != "") el.height = Std.parseInt(element.properties.height);
-					if (element.properties.content_css != "" && element.properties.content_css != null) el.content_css = element.properties.content_css;
+					if (element.properties.contentcss != "" && element.properties.contentcss != null) el.content_css = element.properties.contentcss;
 					
 					el.description = element.properties.description;
 					form.addElement(el);
@@ -569,7 +568,8 @@
 							selectedData.push(Std.string(row.link));
 					}
 					var el = new CheckboxGroup(element.name, label, linkData, selectedData);
-					
+					if (element.properties.formatter != null && element.properties.formatter != "") 
+						el.formatter = Type.createInstance(Type.resolveClass(element.properties.formatter), []);
 					el.description = element.properties.description;
 					form.addElement(el);
 					
@@ -587,6 +587,22 @@
 					var el = new LinkTable(element.name, label, element.properties.table, table, id, null, null, "class=\"resizableFrame\"");
 					el.description = element.properties.description;
 					form.addElement(el);
+					
+				case "enum":
+					var types = application.db.requestSingle("SHOW COLUMNS FROM `"+table+"` LIKE \""+element.properties.name+"\"");
+					
+					var s:String = Reflect.field(types, "Type");
+					var items = s.substr(6, s.length - 8).split("','");
+					
+					var data:List<Dynamic> = new List();
+					for (item in items)
+						data.add( { key:item, value:item } );
+					
+					if (value == null || value == "") value = items[0];
+					
+					var el = new RadioGroup(element.name, label, data, value);
+					el.description = element.properties.description;
+					form.addElement(el);
 			}
 			
 		}
Index: site/cms/modules/base/DatasetItem.mtt
===================================================================
--- site/cms/modules/base/DatasetItem.mtt	(revision 54)
+++ site/cms/modules/base/DatasetItem.mtt	(working copy)
@@ -7,9 +7,6 @@
 	</h3>
 ::end::
 
-
-::raw dbselector::
-
 ::raw form.getErrors()::
 
 ::raw form.getOpenTag()::
Index: site/cms/modules/base/Datasets.hx
===================================================================
--- site/cms/modules/base/Datasets.hx	(revision 54)
+++ site/cms/modules/base/Datasets.hx	(working copy)
@@ -123,7 +123,7 @@
 				
 		} else if (siteMode) { 
 			
-			var pages = application.db.request("SELECT *, p.id as id FROM `_pages` p, `_definitions` d WHERE p.definitionId=d.id ORDER BY d.`order`");
+			var pages = application.db.request("SELECT *, p.id as pid FROM `_pages` p, `_definitions` d WHERE p.definitionId=d.id ORDER BY d.`order`");
 			var tables = application.db.request("SELECT * FROM `_definitions` d WHERE d.isPage='0' ORDER BY `order`");
 			
 			var siteViewData:String = application.db.requestSingle("SELECT `value` FROM `_settings` WHERE `key`='siteView'").value;
Index: site/cms/modules/base/Definition.hx
===================================================================
--- site/cms/modules/base/Definition.hx	(revision 54)
+++ site/cms/modules/base/Definition.hx	(working copy)
@@ -39,7 +39,7 @@
 import php.Web;
 import site.cms.modules.base.Definitions;
 import poko.form.Form;
-import site.cms.components.PopupURL;
+import site.cms.components.DbStructureSelector;
 
 class Definition extends DefinitionsBase
 {	
Index: site/cms/modules/base/DefinitionElement.hx
===================================================================
--- site/cms/modules/base/DefinitionElement.hx	(revision 54)
+++ site/cms/modules/base/DefinitionElement.hx	(working copy)
@@ -153,7 +153,7 @@
 		var data:Dynamic = meta.properties;
 		
 		// for type selector
-		var datatypes:List<Dynamic> = ListData.flatArraytoList(["text", "number", "bool", "image-file", "richtext-tinymce", "richtext-wym", "date", "association", "keyvalue", "read-only", "order", "link-to", "link-value", "hidden"]);
+		var datatypes:List<Dynamic> = ListData.flatArraytoList(["text", "number", "bool", "image-file", "richtext-tinymce", "richtext-wym", "date", "association", "keyvalue", "read-only", "order", "link-to", "link-value", "enum", "hidden"]);
 		
 		// for bool selectors
 		var yesno = new List();
@@ -271,8 +271,15 @@
 		form.addElement(new Input( "def_text_regexError", "Regex Error message", data.regexError), "properties");
 		form.addElement(new Input( "def_text_regexDescription", "Regex Description", data.regexDescription), "properties");
 		form.addElement(new RadioGroup( "def_text_regexCaseInsensitive", "Regex Case Insensitive", yesno, data.regexCaseInsensitive, "0", false), "properties");
+		
+		var textFormatter:Input = new Input( "def_text_formatter", "Formatter Class", data.formatter);
+		textFormatter.useSizeValues = true;
+		textFormatter.width = 400;
+		form.addElement(textFormatter, "properties");
+		
 		form.addElement(new RadioGroup( "def_text_required", "Required", yesno, data.required, "0", false), "properties");
 		
+		
 		form.addElement(new Input( "def_number_min", "Min", data.min), "properties");
 		form.addElement(new Input( "def_number_max", "Max", data.max), "properties");
 		form.addElement(new RadioGroup( "def_number_isInt", "isInt", numberType, data.isInt, "0", false), "properties");
@@ -311,7 +318,7 @@
 		form.addElement(new Selectbox( "def_richtext-tinymce_mode", "Mode", rtf, data.mode, false,""), "properties");
 		form.addElement(new Input( "def_richtext-tinymce_width", "Width", data.width), "properties");
 		form.addElement(new Input( "def_richtext-tinymce_height", "Height", data.height), "properties");
-		form.addElement(new Input( "def_richtext-tinymce_content_css", "CSS file", data.content_css), "properties");
+		form.addElement(new Input( "def_richtext-tinymce_contentcss", "CSS file", data.contentcss), "properties");
 		form.addElement(new RadioGroup( "def_richtext-tinymce_required", "Required", yesno, data.required, "0", false), "properties");
 		
 		form.addElement(new Input( "def_richtext-wym_width", "Width", data.width), "properties");
@@ -391,7 +398,12 @@
 		var assoc:Selectbox = new Selectbox("def_linkdisplay_table", "link Table", tableList, data.table);
 		form.addElement(assoc, "properties");
 		
+		var multiLinkFormatter:Input = new Input( "def_multilink_formatter", "Formatter Class", data.formatter);
+		multiLinkFormatter.useSizeValues = true;
+		multiLinkFormatter.width = 400;
 		
+		form.addElement(multiLinkFormatter, "properties");
+		
 		form.addFieldset("submit", new FieldSet("__submit", "__submit", false));
 		
 		form.setSubmitButton(form.addElement(new Button( "submit", "Submit"), "submit"));

\ No newline at end of file
Index: site/cms/modules/base/js/JsSiteView.hx
===================================================================
--- site/cms/modules/base/js/JsSiteView.hx	(revision 54)
+++ site/cms/modules/base/js/JsSiteView.hx	(working copy)
@@ -32,6 +32,7 @@
 import js.XMLHttpRequest;
 import poko.js.JsRequest;
 import site.cms.modules.base.helper.MenuDef;
+import poko.js.JsUtils;
 
 class JsSiteView extends JsRequest
 {
@@ -50,7 +51,7 @@
 			new JQuery("#addNullButton").click(_t.addNull);
 			
 			_t.flushSorter(null);
-		});
+		});		
 	}
 	
 	private function addSection(e:Event):Void
@@ -272,7 +273,7 @@
 	function editHeading(e:Event)
 	{
 		var t = new JQuery(untyped e.currentTarget).parent().parent().find("p > span");
-		t.html(untyped Lib.prompt("Name? Currently \""+t.html()+"\"."));
+		t.html(untyped JsUtils.prompt("Name? Currently \""+t.html()+"\"."));
 		
 		flushSorter(null);
 		untyped e.preventDefault();
@@ -281,7 +282,7 @@
 	function editItem(e:Event)
 	{
 		var t = new JQuery(untyped e.currentTarget).parent().find("span");
-		t.html(untyped Lib.prompt("Name? Currently \"" + t.html() + "\"."));
+		t.html(untyped JsUtils.prompt("Name? Currently \"" + t.html() + "\"."));
 		
 		flushSorter(null);
 		untyped e.preventDefault();
Index: site/cms/services/Image.hx
===================================================================
--- site/cms/services/Image.hx	(revision 54)
+++ site/cms/services/Image.hx	(working copy)
@@ -51,13 +51,13 @@
 	{
 		var src:String = application.params.get("src");
 		
-		
 		if (application.params.get("preset"))
 		{
 			
 			var image:ImageProcessor = new ImageProcessor(application.uploadFolder + "/" + src);
 			image.cacheFolder = application.uploadFolder+ "/cache";
 			image.format = ImageOutputFormat.JPG;
+			//image.forceNoCache = true;
 			
 			switch(application.params.get("preset"))
 			{
@@ -73,6 +73,8 @@
 					var w:Int = Std.parseInt(application.params.get("w"));
 					var h:Int = Std.parseInt(application.params.get("h"));
 					image.queueFitSize(w, h);
+				case "gallery":
+					image.queueFitSize(10, 10);
 			}
 			
 			var dateModifiedString = DateTools.format(image.dateModified, "%a, %d %b %Y %H:%M:%S") + ' GMT';
			
Index: site/cms/templates/CmsPopup.hx
===================================================================
--- site/cms/templates/CmsPopup.hx	(revision 54)
+++ site/cms/templates/CmsPopup.hx	(working copy)
@@ -31,7 +31,7 @@
 		head.css.add("css/fixes/fonts.css");
 		head.css.add("css/cms/cms.css");
 		
-		head.js.add("js/cms/jquery-1.3.2.min.js");
+		head.js.add("js/cms/jquery.js");
 		head.js.add("js/cms/jquery.domec.js");
 		head.js.add("js/main.js");
 		
